classes/form/CustomerAddressForm.php

```diff
@@ -92,25 +92,25 @@ public function loadAddressById($id_address)
 
     public function fillWith(array $params = [])
     {
-        // This form is very tricky: fields may change depending on which
-        // country is being submitted!
-        // So we first update the format if a new id_country was set.
-        // If detect country from browser language is selected, default country will be overridden
-        if (Tools::isCountryFromBrowserAvailable()) {
-            $languageAvailable = Tools::getCountryIsoCodeFromHeader();
-            $this->formatter->setCountry(new Country(
-                (int) Country::getByIso($languageAvailable, true),
-                Language::getIdByIso($languageAvailable)
-            ));
-        } elseif (isset($params['id_country'])
-            && $params['id_country'] != $this->formatter->getCountry()->id
+        // This form is tricky: fields may change depending on which country is being selected!
+        // Country preselection priority order :
+        // 1) Update the format if a new id_country was set.
+        // 2) Detect country from browser language settings and matches BO enabled countries
+        // 3) Default country set in BO
+
+        if (isset($params['id_country']) && (int) $params['id_country'] !== (int) $this->formatter->getCountry()->id) {
+            $country = new Country($params['id_country'], $this->language->id);
+        } elseif (
+            Tools::isCountryFromBrowserAvailable() &&
+            Country::getByIso($countryIsoCode = Tools::getCountryIsoCodeFromHeader(), true)
```

Comment:
I won't block the PR since it's already been approved and is waiting for QA but the assignment of `$countryIsoCode` is not really a good practice IMHO

It hides the affectation and makes the code less human readable, thus harder to maintain, this should be "split" inside an `else` statement where we assign $countryIsoCode explicitly and then perform the check The logic is good it's just the way code is written which is confusing to me

Reply:
I think @MeKeyCool made the same comment.I get how it can be confusing and as you said maybe less human firendly :P I'll keep it in my mind for future developments